contents/docs/how-posthog-works/queries.mdx

```diff
@@ -0,0 +1,73 @@
+---
+title: Querying data
+---
+
+This page provides an overview of how queries are run when creating insights in PostHog.
+
+> **Note: ** This page is a work in progress and does not cover all the intricacies of how queries are run in PostHog
+
+## Insights counting unique persons
+
+This section covers how PostHog determines the number of unique users who performed a certain action, such as when creating a [Trends](/manual/trends) or [Funnel](/manual/funnel) insight.
+In this case, PostHog determines this number by counting the total number of unique `person_id`'s on events that match your filters.
+
+As an example, let's say that we have the following list of events:
+
+| ID  | Event       | `person_id` | Device ID |
+| --- | ----------- | ----------- | --------- |
+| 1   | viewed page | `user1`     | computer  |
+| 2   | viewed page | `user2`     | laptop    |
+| 3   | viewed page | `user1`     | computer  |
+
+In this case, if we ran a query asking for the number of unique users who viewed a page, we would get a result of `2`, as our table contains 2 unique `person_id`'s.
+
+Now, let's take a look at the same example again, but instead let's say `user1` decided to log in on their phone for the first time.
```

Comment:
I like this approach with tables, but there's a few details here we'll want to adjust for: 
1. mobile libraries never buffer, so we can't fix (4).
2. Let's explain it in terms of initial sign-up flow (with backend server) vs second login.

Furthermore maybe let's move this to how ingestion works and explaining why we do buffering (then in here just say due to the way we write person_ids we can sometimes see higher numbers in the insight, see more and link to ingestion pipeline where we explain this)

I propose something like this

initial events example:
```
| ID  | Event       | `distinct-id` | what person-id it will get during ingestion |
| --- | ----------- | ----------- | --------|
| 1   | viewed page | `anon-1`     | `user-1` |
| 2   | viewed page | `anon-2`     | `user-2` |
| 3   | viewed page | `anon-1`     |  `user-1` |
```
Here we can say we have two different users viewing the page.

Secondly let's look at a sign-up flow (identify event is used to sign-up/log-in)
```
| ID  | Event       | `distinct-id` | anon-distinct-id | what person-id it will get during ingestion
| --- | ----------- | ----------- | --------|
| 1   | viewed page | `anon-1`     | null | `user-1`
| 2   | identify | `user@email.com`     | `anon-1`  | `user-1`
| 3   | viewed page | `anon-1`     |  null | `user-1`
| 4   | viewed page | `user@email.com`     |  null | `user-1`
```
This is the example for when we saw an identify call and only one of the distinct id's has a user attached to it.

sign-up flow with some backend events (if we didn't use the buffer)
```
| ID  | Event       | `distinct-id` | anon-distinct-id | what person-id it will get during ingestion
| --- | ----------- | ----------- | --------|
| 1   | viewed page | `anon-1`     | null | `user-1`
| 2  | backend sign-up event | `user@email.com` | null | `user-2`  - the user didn't exist, so we create it
| 2   | identify | `user@email.com`     | `anon-1`  | `user-2` - both exist, use identified one
| 3   | viewed page | `anon-1`     |  null | `user-2` - now from the previous one they are linked both to `user-2`
| 4   | viewed page | `user@email.com`     |  null | `user-2`
```

But due to the buffer we delay the backend event, so the events are ingested in perhaps an order like this:
```
| ID  | Event       | `distinct-id` | anon-distinct-id | what person-id it will get during ingestion
| --- | ----------- | ----------- | --------|
| 1   | viewed page | `anon-1`     | null | `user-1`
| 2   | identify | `user@email.com`     | `anon-1`  | `user-1` - we can just associate as identified doesn't exist yet
| 2  | backend sign-up event | `user@email.com` | null | `user-1`  - person exists this is just a normal event
| 3   | viewed page | `anon-1`     |  null | `user-1` 
| 4   | viewed page | `user@email.com`     |  null | `user-1`
```

When the buffer can't help us and we don't do anything else either (we really care about the inital sign-up flow more, to know where your users came from, but second/third/... sessions events before logins aren't usually very informative, so we've decided to not introduce more complexity here, which would risk reliability)

```
| ID  | Event       | `distinct-id` | anon-distinct-id | what person-id it will get during ingestion
| --- | ----------- | ----------- | --------|
| 1   | viewed page | `anon-1`     | null | `user-1`
| 2   | identify | `user@email.com`     | `anon-1`  | `user-1` - we can just associate as identified doesn't exist yet
| 2  | viewed page | `user@email.com` | null | `user-1`  - person exists this is just a normal event
| 3   | viewed page | `anon-2`     |  null | `user-2`  - imagine this was the same user just the next day opened the page before logging in
| 2   | identify | `user@email.com`     | `anon-2`  | `user-1` - both exist - associate with identified person as that history is likely more important
| 4   | viewed page | `user@email.com`     |  null | `user-1`
| 3   | viewed page | `anon-2`     |  null | `user-1`  - future events will get associated with `user-2`
```

nit change all `viewed page` to `pageview` (as that's the name of the event we have so would feel more natural)

Reply:
here we'll likely want to call out what we'll be doing in the UI based on https://github.com/PostHog/posthog/issues/11488